include/auxvariables/auxiliary_variables.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+#include <memory>
+#include <iostream>
+#include <fstream>
+#include "mesh_extras.hpp"
+
+
+namespace hephaestus
+{
+
+class AuxiliaryVariable
+{
+    public:
+    AuxiliaryVariable();
+    AuxiliaryVariable(const std::string& var_name, 
+                      const std::string& var_family,
+                      int var_order,
+                      mfem::FiniteElementSpace& fespace);
+    
+    std::string name;
+    std::string family;
+    int order;
+    mfem::GridFunction gf;
+};
+
+
+class AuxVarMap : public std::map<std::string, hephaestus::AuxiliaryVariable *>{
```

Comment:
Will likely be made redundant by having named lists of variables in Hephaestus down the line, as it saves making a new class - but good for now